### PR TITLE
Add yv healthcheck

### PR DESCRIPTION
**Pull request by dragospitica**

### CODE REVIEW

File: src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatistics.java [hunk 1]
```diff
@@ -13,12 +13,21 @@
...
+    private final int failedTasks;
     
-    public FeedStatistics(Publisher publisher, int queueSize, 
-            Duration updateLatency) {
+    public FeedStatistics(
+            Publisher publisher,
+            int queueSize,
+            Duration updateLatency,
+            int createdTasks,
+            int failedTasks
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatistics.java]: Use a builder with this many parameters

File: src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatistics.java [hunk 2]
```diff
@@ -13,12 +13,21 @@
     private final Publisher publisher;
     private final int queueSize;
     private final Duration updateLatency;
+    private final int createdTasks;
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatistics.java]: Both of these names are wrong, given they're "created" and "failed" in some delta time. You need to either store the `Duration` that the query had or name these better.

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 3]
```diff
@@ -77,4 +84,29 @@ private int calculateCurrentQueueSize(Publisher publisher) {
...
+
+        return collection.find(query).count();
+    }
+
+    private int getTasksFailedInTheLastFourHours() {
+        Date fourHoursAgo = DateTime.now().minusHours(4).toDate();
+
+        DBObject query = QueryBuilder.start()
+                .put("created").greaterThanEquals(fourHoursAgo)
+                .and("status").is("REJECTED")
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: There is also a status `FAILED`, have you checked the difference and which ones we want here?

File: src/main/java/org/atlasapi/feeds/youview/statistics/simple/FeedStatistics.java [hunk 4]
```diff
@@ -46,12 +48,30 @@ public String getUpdateLatencyString() {
         return updateLatencyString;
     }
 
+    public int getCreatedTasks() {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/simple/FeedStatistics.java]: The convention in this class for getters is to not have the `get` prefix.

File: src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatistics.java [hunk 5]
```diff
@@ -59,4 +82,50 @@ public boolean equals(Object that) {
...
+            return this;
+        }
+
+        public Builder withUnsuccessfulTasks(int unsuccessfulTasks) {
+            this.unsuccessfulTasks = unsuccessfulTasks;
+            return this;
+        }
+
+        public FeedStatistics createFeedStatistics() {
+            return new FeedStatistics(
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatistics.java]: Use the new builder style. Call the create method build, and pass the builder into the constructor

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 6]
```diff
@@ -77,4 +104,53 @@ private int calculateCurrentQueueSize(Publisher publisher) {
...
+        }
+
+        public Builder withDestinationType(
+                Destination.DestinationType destinationType) {
+            this.destinationType = destinationType;
+            return this;
+        }
+
+        public MongoFeedStatisticsStore build() {
+            return new MongoFeedStatisticsStore(mongo, taskStore, clock, destinationType);
```
Comment (AdamDM) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: same, pass the builder object in

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 7]
```diff
@@ -30,24 +34,47 @@
...
     }
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
     @Override
-    public Optional<FeedStatistics> resolveFor(Publisher publisher) {
+    public Optional<FeedStatistics> resolveFor(Publisher publisher, java.time.Duration duration) {
+        int successfulTasks = getTasksInTheLastFourHours(duration, "ACCEPTED", "PUBLISHED");
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: The method name is wrong. It's not `LastFourHours` any more.

File: src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatisticsResolver.java [hunk 8]
```diff
@@ -1,11 +1,12 @@
...
+
 import org.atlasapi.media.entity.Publisher;
 
 import com.google.common.base.Optional;
 
-
 public interface FeedStatisticsResolver {
 
-    Optional<FeedStatistics> resolveFor(Publisher publisher);
+    Optional<FeedStatistics> resolveFor(Publisher publisher, Duration duration);
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatisticsResolver.java]: The argument name needs fixing, `duration` carries no semantic meaning.

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 9]
```diff
@@ -30,24 +34,47 @@
...
         this.destinationType = checkNotNull(destinationType);
     }
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
     @Override
-    public Optional<FeedStatistics> resolveFor(Publisher publisher) {
+    public Optional<FeedStatistics> resolveFor(Publisher publisher, java.time.Duration duration) {
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: Argument name.

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 10]
```diff
@@ -77,4 +104,53 @@ private int calculateCurrentQueueSize(Publisher publisher) {
 
         return Optional.of(new Duration(oldestMessage, clock.now()));
     }
+
+    private int getTasksInTheLastFourHours(java.time.Duration duration, String taskStatus1, String taskStatus2) {
+        DBObject firstClause = QueryBuilder.start("status").is(taskStatus1).get();
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: Both `firstClause` and `secondClause` are bad names, as are `taskStatus1` and `taskStatus2`. The method name also doesn't match what it's doing.

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 11]
```diff
@@ -77,4 +104,53 @@ private int calculateCurrentQueueSize(Publisher publisher) {
...
+        DBObject secondClause = QueryBuilder.start("status").is(taskStatus2).get();
+
+        int hours = (int) duration.toHours();
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: Dafuq. This is completely unnecessary, Joda has a `.minus(Duration)` method. You potentially want to take in a Joda `Duration`, not a java time one for that.

File: src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatisticsResolver.java [hunk 12]
```diff
@@ -3,9 +3,9 @@
 import org.atlasapi.media.entity.Publisher;
 
 import com.google.common.base.Optional;
-
+import org.joda.time.Period;
 
 public interface FeedStatisticsResolver {
 
-    Optional<FeedStatistics> resolveFor(Publisher publisher);
+    Optional<FeedStatistics> resolveFor(Publisher publisher, Period requiredPeriod);
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/FeedStatisticsResolver.java]: `requiredPeriod`..? Required for what? Something like `timeBeforeNow` would be much more obvious.

File: src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java [hunk 13]
```diff
@@ -30,24 +35,50 @@
...
+    public static Builder builder() {
+        return new Builder();
     }
 
     @Override
-    public Optional<FeedStatistics> resolveFor(Publisher publisher) {
+    public Optional<FeedStatistics> resolveFor(Publisher publisher, Period requiredPeriod) {
+        int successfulTasks = getTasksCreatedInTheLastDurationByStatus(
+                requiredPeriod,
+                "ACCEPTED",
```
Comment (emilssolmanis) [inline src/main/java/org/atlasapi/feeds/youview/statistics/MongoFeedStatisticsStore.java]: Also isn't the `Status` enum available here? Why are we using magic strings?

**Pull request merged**